### ISSUE TITLE
Cycle-GAN training

### ISSUE THREAD

**Issue by IssamLaradji** [gjy3035/GCC-SFCN#5]
Hii! Does the code have the cycle-gan part of the paper? Thanks!

**Comment by gjy3035** [comment]
For SE Cycle GAN, we add a loss in Cycle GAN during the reconstruction process. 
In practice, we add a .py file and modify the model file. 

Considering the minor improvement based on Cycle GAN, we provide some related files. 
[SE CycleGAN.zip](https://github.com/gjy3035/GCC-SFCN/files/2974663/SE.CycleGAN.zip)

pytorch_ssim: loss computation.
cycle_gan_model.py: SE Cycle GAN model file
filter.py: Scene Regulization.

**Comment by gjy3035** [comment]
Because of many deadlines in March and April, we don't have enough time to check the entire SE Cycle GAN code and release it.  After April, we will try our best to open source it ASAP. 
If you have any problems during training SE Cycle GAN, you can submit issues or send email to me: gjy3035@gmail.com. Thanks for your attention!

**Comment by IssamLaradji** [comment]
Thanks a lot @gjy3035  that's very helpful! Would it be okay if you provide the hyperparameters you used with Cycada/SSIM?

For example,
- the lambda in the reconstruction loss;
- the crop size of the input image;
- the batch size; and
- the network architectures of the discriminators and generators?

Thanks a lot for your engagement! :)

**Comment by gjy3035** [comment]
Other settings are followed by [pix2pix's](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/tree/pytorch0.3.1) (pytorch 0.3.1) default configuration  (the paper also mentions it). To be specific,
- lambda is 10.0;
- size setting:
  - pre process: resize GCC to 540*960
  - training: crop size 360*360
- batch size is 2 on two 1080Ti GPUs (you may adopt a larger value on Titan V or other high-performance devices);
- same as the default selection: basic and resnet_9blocks

Here, we provide the opt.txt for you. 
[opt.txt](https://github.com/gjy3035/GCC-SFCN/files/2976035/opt.txt)
Two key setting:
`loadSize: 540`: height size for pre processing;
`resize_or_crop: scale_height_and_crop`: keeping the ratio and resize GCC to height of 540, i.e., the entire resolution is 540*960 (you may add some code similar to "scale_width_and_crop" in `data/base_dataset.py`).

**Comment by CommissarMa** [comment]
I wonder if the process of generating realistic GCC images are as follows:  
1. use 'scale_height_and_crop' to get 360*360 patches for CycleGAN training as your opt.txt above.
2. use trained netG to get realistic GCC images with 540*960.
Thank you.
